docker-compose.yml

```
@@ -107,9 +116,20 @@ services:
     restart: always
     volumes:
       - ./pong4-server/game-server:/app
+    networks:
+      - backend-network
 
   redis:
     image: redis:7
     restart: always
+    networks:
+      - backend-network
   # ports:
-  #   - "6379:6379"
\ No newline at end of file
+  #   - "6379:6379"
+
+networks:
+  backend-network:
+    driver: bridge
+  frontend-network:
+    name: frontend-network
+    external: true
```

`frontend-network` の作成を本体側にすると良さそうです
(先に本体側を立ち上げると`Error response from daemon: network frontend-network not found`になる)

```suggestion
    driver: bridge
    ipam:
      config:
        - subnet: 172.20.0.0/16
    external: false
```

test-e2eのdocker-compose.ymlではすでにあるものを使用するように指定
```
-    driver: bridge
-    ipam:
-      config:
-        - subnet: 172.20.0.0/16
-    external: false
---
+    external: true
```

---

ありがとうございます！
あ、netoworkを一度手動で立ち上げた状態で動かしていたので問題が露見しなかったようです。

セキュリティホールが出来そうで本体側（サーバー側）で穴の空いたネットワークを作りたくないという理由からテスト側（test-e2e側）で生成していたんですが、考えてみたら本番用とテスト時用のymlを分ければよいだけなことに気が付きました。
あとで修正入れます！